subsys/net/ip/tcp.c

```diff
@@ -379,6 +379,8 @@ static int prepare_segment(struct net_tcp *tcp,
 	struct net_context *context = tcp->context;
 	struct net_buf *tail = NULL;
 	struct net_tcp_hdr *tcp_hdr;
+	struct net_if *interface;
```

Comment:
Why these are introduced here, if they are used only in the `else {}` block (and thus can be very well declared there?).